internal/workflow.go

```diff
@@ -46,92 +46,229 @@ var (
-	//	}
-	// should be written as
-	// 	s := workflow.NewSelector(ctx)
-	//	counter := 0
-	//	s.AddReceive(workflow.GetSignalChannel(ctx, "channelA"), func(c workflow.Channel, ok bool) {
-	//		if ok{
-	//			var i int
-	//			c.Receive(ctx, &i)
-	//			counter += i
-	//		}
-	//	})
-	//	s.AddReceive(workflow.GetSignalChannel(ctx, "channelB"), func(c workflow.Channel, ok bool) {
-	//		if ok{
-	//			var i int
-	//			c.Receive(ctx, &i)
-	//			counter += i
-	//		}
-	//	})
+	// Selector must be used in workflows instead of a native Go select statement.
+	//
+	// Use workflow.NewSelector(ctx) to create a Selector instance, and then add cases to it with its methods.
+	// The interface is intended to simulate Go's select statement, and any Go select can be fairly trivially rewritten
+	// for a Selector with effectively identical behavior.
+	//
+	// For example, normal Go code like below:
+	//  chA := make(chan int)
+	//  chB := make(chan int)
+	//  counter := 0
+	//  for {
+	//  	select {
+	//  	case i, more := <- chA:
+	//  		if more {
+	//  			counter += i
+	//  		}
+	//  	case i, more := <- chB:
+	//  		if more {
+	//  			counter += i
+	//  		}
+	//  	case <- time.After(time.Hour):
+	//  		break
+	//  	}
+	//  }
+	// can be written as:
+	//  chA := workflow.NewChannel(ctx)
+	//  chB := workflow.NewChannel(ctx)
+	//  counter := 0
+	//  for {
+	//  	timedout := false
+	//  	s := workflow.NewSelector(ctx)
+	//  	s.AddReceive(chA, func(c workflow.Channel, more bool) {
+	//  		if more {
+	//  			var i int
+	//  			c.Receive(ctx, &i)
+	//  			counter += i
+	//  		}
+	//  	})
+	//  	s.AddReceive(chB, func(c workflow.Channel, more bool) {
+	//  		if more {
+	//  			var i int
+	//  			c.Receive(ctx, &i)
+	//  			counter += i
+	//  		}
+	//  	})
+	//  	s.AddFuture(workflow.NewTimer(ctx, time.Hour), func(f workflow.Future) {
+	//  		timedout = true
+	//  	})
+	//  	s.Select(ctx)
+	//  	if timedout {
+	//  		break
+	//  	}
+	//  }
+	//
+	// You can create a new Selector as needed or mutate one and call Select multiple times, but note that:
+	//
+	// 1. AddFuture will not behave the same across both patterns.  Read AddFuture for more details.
 	//
-	//	for {
-	//		s.Select(ctx)
-	//	}
+	// 2. There is no way to remove a case from a Selector, so you must make a new Selector to "remove" them.
+	//
+	// Finally, note that Select will not return until a condition's needs are met, like a Go selector - canceling the
+	// Context used to construct the Selector, or the Context used to Select, will not (directly) unblock a Select call.
+	// Read Select for more details.
 	Selector interface {
-		// AddReceive adds a ReceiveChannel to the selector. f is invoked when the channel has data or closed.
-		// ok == false indicates the channel is closed
-		AddReceive(c Channel, f func(c Channel, ok bool)) Selector
-		// AddSend adds a SendChannel to the selector. f is invoke when the channel is available to send
+		// AddReceive waits to until a value can be received from a channel.
```

Comment:
maybe 
`AddReceive adds a callback f, to wait and invoke f until a value can be received from a channel`

Reply:
actually, I don't like mine(too hard to read). So feel free to keep it.

Reply:
looks like I did screw up the text here in any case.  will fix 👍